src/main/java/com/google/javascript/clutz/DeclarationGenerator.java

```diff
@@ -1382,21 +1352,63 @@ void walkInnerSymbols(ObjectType type, String innerNamespace) {
           privateEnums.add(innerNamespace + '.' + propName);
         }
       }
```

Comment:
I hate this pattern, but this is the only sure way to emit only when needed. Otherwise the logic is too complicated. Any ideas how to clean it up?